lib/active_record_upsert/arel/crud.rb

```diff
@@ -12,9 +12,11 @@
 module Arel
   module Crud
     def compile_upsert(upsert_keys, upsert_options, upsert_values, insert_values, wheres)
+      # Support non-attribute key (like `md5(my_attribute)``)
+      target = self[upsert_keys.class == Hash ? ::Arel::Nodes::SqlLiteral.new(upsert_keys[:literal]) : upsert_keys.join(',')]
```

Comment:
Is there a reason to do `upsert_keys.class == Hash` instead of something like `upsert_keys.is_a? Hash`? In rails-land we might encounter something like a `HashWithIndifferentAccess`

Reply:
So a `kind_of?` should do the trick too